docs/src/pages/components/checkboxes/CustomizedCheckbox.tsx

```diff
@@ -1,63 +1,56 @@
 import * as React from 'react';
-import clsx from 'clsx';
-import { makeStyles } from '@material-ui/core/styles';
+import { experimentalStyled as styled } from '@material-ui/core/styles';
 import Checkbox, { CheckboxProps } from '@material-ui/core/Checkbox';
 
-const useStyles = makeStyles({
-  root: {
-    '&:hover': {
-      backgroundColor: 'transparent',
-    },
+const Icon = styled('span')({
+  borderRadius: 3,
+  width: 16,
+  height: 16,
+  boxShadow: 'inset 0 0 0 1px rgba(16,22,26,.2), inset 0 -1px 0 rgba(16,22,26,.1)',
+  backgroundColor: '#f5f8fa',
+  backgroundImage: 'linear-gradient(180deg,hsla(0,0%,100%,.8),hsla(0,0%,100%,0))',
+  '.root.Mui-focusVisible &': {
+    outline: '2px auto rgba(19,124,189,.6)',
+    outlineOffset: 2,
   },
-  icon: {
-    borderRadius: 3,
+  'input:hover ~ &': {
+    backgroundColor: '#ebf1f5',
+  },
+  'input:disabled ~ &': {
```

Comment:
Let's never omit the CSS selector, for instance, here


```suggestion
  '& input:disabled ~ &': {
```

Reply:
Resolved

Reply:
@oliviertassinari This way, the disabled style stops working.

https://www.argos-ci.com/mui-org/material-ui/builds/1271